File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows==1 or numRows>= len(s):
            return s

        l = []
        for i in range(numRows):
            l.append([])
        row=0
        result = ''
        flag=True

        for c in s:
            l[row].append(c)

            if row==0:
                flag=True
            elif row == numRows-1:
                flag=False
            
            if flag:
                row+=1
            else:
                row-=1

        for c in l:
            result += ''.join(c)

        return result

        

File: 0006-zigzag-conversion/0006-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows==1 || numRows>= s.size()){
            return s;
        }
        vector<string> vec(numRows);
        int row=0;
        bool flag=true;

        for(char c: s){
            vec[row].push_back(c);

            if(row==0){
                flag=true;
            }
            else if(row==numRows-1){
                flag=false;
            }

            if(flag){
                row++;
            }
            else{
                row--;
            }
        }
        string result;
        for(string&r : vec){
            result += r;
            }
        
        return result;
        
    }   
};